op_crates/fetch/lib.rs

```diff
@@ -156,7 +157,10 @@ where
   let status = res.status();
   let mut res_headers = Vec::new();
   for (key, val) in res.headers().iter() {
-    res_headers.push((key.to_string(), val.to_str().unwrap().to_owned()));
+    res_headers.push((
+      key.to_string(),
+      from_utf8(val.as_bytes()).unwrap().to_owned(),
```

Comment:
The `.unwrap()` will kill the process when the header is not valid UTF-8 (e.g., because it's Latin-1.)

You could try to parse the byte string as UTF-8 first, Latin-1 second, but some Latin-1 strings are also legal (if not legible) UTF-8 strings, and of course all UTF-8 strings are legal Latin-1. Occasional mojibake is inevitable with that approach.

For reference, Node.js has been parsing headers as Latin-1 from the moment V8 grew a concept of non-UTF-8 strings (commit nodejs/node@f674b09f40d22915e15b6968aafc5d25ac8178a2 from August 2013.)

Reply:
@bnoordhuis Thank you for your feedback!

I wasn't thinking straight. Checking your comments and code of headers, I've understood that the current implementation supports Latin-1.

And i also found out that the validator doesn't allow the headers value to be parsed correctly as UTF-8.

However, if we keep the existing code line 159, the process will be killed by `unwrap()` as in the above issue. And the explanation for the cause seems lacking.

So It seems good to solve the above issue by showing the detailed reason for the occurrence of an non-ascii error for` to_str.`

What do you think about this?